pages/blog/react-as-generator-engine.md

```diff
@@ -0,0 +1,221 @@
+    return this.props.childrenContent;
+  }
+}
+
+function FunctionComponent() {
+  return (
+    <Indent size={3} type={IndentationTypes.TABS}>
+      indented text
+      <ClassComponent>
+        <Text newLines={2}>
+          text wrapped by custom component
+        </Text>
+      </ClassComponent>
+    </Indent>
+  );
+}
+
+// content will be `\t\t\tindented text text wrapped by custom component\n\n`
+const content = render(<FunctionComponent />);
+```
+
+Unfornatelly we have some restrictions:
+
+- React hooks is not allowed.
+- HTML tags at the moment is not supported.
+- React internal components like `Fragments`, `Suspense` etc. are skipped.
+
+## Comparison with Nunjucks
+
+If you don't know, the existing templates developers were forced (if they used our Generator) to write in [Nunjucks](https://mozilla.github.io/nunjucks/). It's a templating language, heavily focused on string literals, filters (similar to bash pipes) and partials called macros.
+
+Below are some examples comparing doing those same things in Nunjucks and React. If you want check another, more complex, examples please see [`template-for-generator-templates`](https://github.com/asyncapi/template-for-generator-templates) repository and coresponding [Nunjucks](ttps://github.com/asyncapi/template-for-generator-templates/tree/nunjucks) branch.
+
+### Make reusable parts
+
+It may sound trivial, but when writing any code, even a template, a programmer wants to create reusable parts that will separate the given, very frequent repeating logic.
+
+As we mentioned, in Nunjucks partials are called macros, in React otherwise we called those parts as `components` (we follow with React's nomenclature). Imagine that you are writing a template that produces a Markdown's content. You need to create a reusable macro/component that will be able to render a list based on given an array of strings.
+
+For example, using Nunjucks you can write something like:
+
+```njk
+{# partials/list.njk #}
+
+{% macro list(data, type = "-") %}
+{% for item in data %}
+{{type}} {{item}}
+{% endfor %}
+{% endmacro %}
+
+{# template/Readme.md #}
+
+{% from "partials/list.njk" import list %}
+{{ list(["one", "two", "three"]) }}
+```
+
+In React we can write it in this way:
+
+```js
+function List({ list = [], type = "-" }) {
+  return list.map(item => `${type} ${item}\n`);
+}
+
+// use `List` component in another component
+export function SimpleList() {
+  return (
+    <List list={["one", "two", "three"]} />
+  );
+}
+```
+
+Looking at both examples we see that in Nujucks we operate on string literals, it means that when passing data to a macro, you always need to know what type of data the macro takes. In React we operate on JS objects/variables. By this, your IDE should always inform you what value, of what type, you must pass to component. Also, Nunjucks's macro should be created inside `partials` folder of template. Using React, you can create component whenever you want.
```

Comment:
```suggestion
Looking at both examples we see that in Nujucks we operate on string literals, it means that when passing data to a macro, you always need to know what type of data the macro takes. In React we operate on JS objects/variables. By this, your IDE should always inform you what value, of what type, you must pass to component. Also, Nunjucks's macro should be created inside `.partials` folder of template. Using React, you can create component whenever you want.
```
In the generator we list it as `.partials` 😄 https://github.com/asyncapi/generator/blob/master/docs/authoring.md#partials